tap_hubspot/hubspot.py

```diff
@@ -238,55 +248,44 @@ def get_contacts(
             MANDATORY_PROPERTIES[f"{obj_type}"],
         )
 
-        while True:
-            deals = {}
-            i = 0
-
-            for rec, ts in gen:
-                rec_id = rec["id"]
-                # ensure that there is a default entry for all contacts/companies
-                rec["associations"] = {
-                    # fit the legacy format to be backwards compatible:
-                    # "results": [{ "id": "<id>" }]
-                    "deals": {"results": []},
-                    "companies": {"results": []},
+        for chunk in chunker(gen, 10):
+            ids: List[str] = [contact["id"] for contact in chunk]
+
+            deals_associations = self.get_associations(obj_type, "deals", ids)
+            companies_associations = self.get_associations(obj_type, "companies", ids)
+            for i, deal_id in enumerate(ids):
+                contact = chunk[i]
+                deals = deals_associations.get(deal_id, [])
+                companies = companies_associations.get(deal_id, [])
+                contact["associations"] = {
+                    "deals": {"results": deals},
+                    "companies": {"results": companies},
                 }
-
-                deals[rec_id] = rec
-                i += 1
-                if i >= 10:
-                    break
-
-            ids = list(deals.keys())
-
-            for id, ass in self.get_associations(obj_type, "deals", ids):
-                deals[id]["associations"]["deals"] = {"results": ass}
-
-            for id, ass in self.get_associations(obj_type, "companies", ids):
-                deals[id]["associations"]["companies"] = {"results": ass}
-
-            for deal in deals.values():
-                yield deal, parser.isoparse(
-                    self.get_value(deal, ["properties", filter_key])
+                yield contact, parser.isoparse(
+                    self.get_value(contact, ["properties", filter_key])
                 )
 
+
     def get_associations(
         self,
         from_obj: str,
         to_obj: str,
         ids: List[str],
-    ) -> Iterable[Tuple[str, List[Dict[str, str]]]]:
+    ) -> Dict[str, List[Dict[str, str]]]:
         body = {"inputs": [{"id": id} for id in ids]}
-        path = f"/crm/v3/associations/{from_obj.title()}/{to_obj.title()}/batch/read"
+        path = f"/crm/v3/associations/{from_obj}/{to_obj}/batch/read"
```

Comment:
this PR should only contain changes from the get_contacts functions, all the helper functions is going to create work for you to clean up later.